vividus-plugin-web-app/src/main/java/ru/yandex/qatools/ashot/shooting/DebuggingViewportPastingDecorator.java

```diff
@@ -39,11 +39,27 @@
  */
 // CHECKSTYLE:OFF
 public class DebuggingViewportPastingDecorator extends ShootingDecorator {
+    private static final String SCROLL_Y_TEMPLATE =
+              "if ('scrollBehavior' in document.documentElement.style) {"
+            + "    %1$s.scrollTo({"
+            + "        \"top\": arguments[0],"
+            + "        \"left\": 0,"
+            + "        \"behavior\": \"instant\""
+            + "    });"
+            + "} else {"
+            + "    %1$s.scrollTo(0, arguments[0]);"
+            + "}"
+            + "return [];";
     private static final long serialVersionUID = 4173686031614281540L;
     private static final String CURRENT_SCROLL = "current_scroll_";
     protected int scrollTimeout = 0;
     private Coords shootingArea;
     private transient ScreenshotDebugger screenshotDebugger;
+    protected final String scrollYScript;
+
+    {
+        scrollYScript = String.format(SCROLL_Y_TEMPLATE, getTarget());
```

Comment:
what about moving this part to constructor and converting `getTarget()` to a constructor parameter?